<launch>

  <arg name="on_amiga" default="true"/>

  <!-- realsense -->
  <include file="$(find realsense2_camera)/launch/rs_aligned_depth.launch">
    <arg name="enable_pointcloud" value="true" />
  </include>

  <!-- aruco ros -->
  <arg name="markerSize"      default="0.09"/>    <!-- in m -->
  <arg name="side"             default="left"/>
  <node pkg="aruco_ros" type="marker_publisher" name="aruco_marker_publisher">
      <remap from="/camera_info" to="/camera/color/camera_info" />
      <remap from="/image" to="/camera/color/image_raw" />
      <param name="image_is_rectified" value="false"/>
      <param name="marker_size"        value="$(arg markerSize)"/>
      <param name="reference_frame"    value="camera_color_optical_frame"/>
      <param name="camera_frame"       value="camera_color_optical_frame"/>
  </node>

  <!-- tf node-->
  <node pkg="peter"  name="tf_handler_node" type="tf_handler_node.py"></node>

  <!-- state machine node-->
  <node pkg="peter"  name="state_machine_node" type="state_machine_node.py"></node>

  <!-- planner node-->
  <node pkg="peter"  name="planner_node" type="planner_node.py"></node>

  <!-- ee node-->
  <node pkg="peter"  name="ee_node" type="ee_node.py"></node>

  <!-- perception node-->
  <node pkg="peter"  name="perception_node" type="perception_node.py"></node>

  <!-- manipulation node -->
  <node pkg="peter"  name="manipulator" type="manipulator.py"></node>

  <!-- pepper filter node -->
  <node pkg="peter"  name="pepper_filter_node" type="pepper_filter_node.py"></node>

  <!-- static tf from world to link_base -->
  <node pkg="tf2_ros" type="static_transform_publisher" name="rviz_tf" args="0 0 0 1 0 0 0  world link_base" if="$(arg on_amiga)"/>
  
  <!-- gui node -->
  <node pkg="peter"  name="pepper_gui" type="gui_backend.py"></node>

  <!-- depth smoothing node -->
  <node pkg="peter" name="depth_smoother" type="depth_smoothing.py"></node>

  <!-- amiga aruco node -->
  <node pkg="peter" name="amiga_aruco_node" type="amiga_aruco.py"></node>

</launch>